src/table/components/head/ColumnAdjuster.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useRef } from 'react';
+import { AdjusterProps } from '../../types';
+import { useContextSelector, TableContext } from '../../context';
+import { AdjusterHitArea, AdjusterHeadBorder, AdjusterBodyBorder } from './styles';
+import { ColumnWidthTypes, MIN_COLUMN_WIDTH, PAGINATION_HEIGHT } from '../../constants';
+
+/**
+ * Component that is placed on top of column border.
+ * The vertical borders in the header can be used to change the column width.
+ * When you start dragging, mouse move and mouse up listeners are added.
+ * While dragging the current column is updated, and on mouse up all other columns are updated.
+ */
+const ColumnAdjuster = ({ column, isLastColumn }: AdjusterProps) => {
```

Comment:
Not sure how to test this properly. Simulating mousedown, mousemove and mouse up separately seems very flaky to me. If anything I will look at it in a different PR

Reply:
One alternative for testing this logic would be to wrap the logic itself in a custom hook. Test the result of that hook. Ex. it returns the handlers, your test calls those with a fake even and you verify some outcome of that.

Reply:
yeah It thought about that, we do it like that for selections, but they are actually just non-react functions. I will try that in the next pr